### PR TITLE
chore: Update docs with new release workflow

### PR DESCRIPTION
**Pull request by kevjumba**
Signed-off-by: Kevin Zhang <kzhang@tecton.ai>

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

### CODE REVIEW

File: docs/project/release-process.md [hunk 1]
```diff
@@ -4,60 +4,22 @@
...
---max-issues 1 \
--o
-```
-
-1. Review each change log item.
-   * Make sure that sentences are grammatically correct and well formatted (although we will try to enforce this at the PR review stage). 
-   * Make sure that each item is categorised correctly. You will see the following categories: `Breaking changes`, `Implemented enhancements`, `Fixed bugs`, and `Merged pull requests`. Any unlabelled PRs will be found in `Merged pull requests`. It's important to make sure that any `breaking changes`, `enhancements`, or `bug fixes` are pulled up out of `merged pull requests` into the correct category. Housekeeping, tech debt clearing, infra changes, or refactoring do not count as `enhancements`. Only enhancements a user benefits from should be listed in that category.
-   * Make sure that the "Full Change log" link is actually comparing the correct tags (normally your released version against the previously version).
-   * Make sure that release notes and breaking changes are present.
+1. Merge upstream master changes into your fork.
```
Comment (achals) [inline docs/project/release-process.md:7]: Maybe this should have two sections, one on how to run the `build_wheels` workflow on personal forks with a manually applied tag to verify wheels are being created correctly, and a second on how to run the `release` workflow using your personal access token on master in the feast repo.

Comment (kevjumba) [reply docs/project/release-process.md:9]: fixed

File: docs/project/release-process.md [hunk 2]
```diff
@@ -4,60 +4,27 @@
...
---max-issues 1 \
--o
-```
-
-1. Review each change log item.
-   * Make sure that sentences are grammatically correct and well formatted (although we will try to enforce this at the PR review stage). 
-   * Make sure that each item is categorised correctly. You will see the following categories: `Breaking changes`, `Implemented enhancements`, `Fixed bugs`, and `Merged pull requests`. Any unlabelled PRs will be found in `Merged pull requests`. It's important to make sure that any `breaking changes`, `enhancements`, or `bug fixes` are pulled up out of `merged pull requests` into the correct category. Housekeeping, tech debt clearing, infra changes, or refactoring do not count as `enhancements`. Only enhancements a user benefits from should be listed in that category.
-   * Make sure that the "Full Change log" link is actually comparing the correct tags (normally your released version against the previously version).
-   * Make sure that release notes and breaking changes are present.
+### Pre-release Verification
```
Comment (felixwang9817) [inline docs/project/release-process.md:7]: can you add a note on what kind of verification we're doing here? (e.g. just building wheels without publishing)

File: docs/project/release-process.md [hunk 3]
```diff
@@ -4,60 +4,27 @@
...
+
+1. Merge upstream master changes into your fork.
+2. Create a tag manually for the release on your fork. For example, if your release is on version 0.22.0, create a tag by doing the following.
```
Comment (felixwang9817) [inline docs/project/release-process.md:10]: "if your release is on version 0.22.0" is a bit confusing to me - maybe something like "if you are doing a release for version 0.22.0"?

File: docs/project/release-process.md [hunk 4]
```diff
@@ -4,60 +4,27 @@
...
+   - Checkout master branch and run `git tag v0.22.0`.
+   - Run `git push --tags` to push the tag to remote.
+3. Access the `Actions` tab on your github UI on your fork and click the `build_wheels` action.
+4. Look for the header `This workflow has a workflow_dispatch event trigger` and click `Run Workflow` on the right.
+5. Run the branch off of the tag you just created(`v0.22.0` in this case) and verify that the workflow worked.
```
Comment (felixwang9817) [inline docs/project/release-process.md:15]: don't we need to delete this tag afterwards?

Comment (kevjumba) [reply docs/project/release-process.md:15]: Not really? Since presumably it would work and the next release cycle we would add the new tag.

File: docs/project/release-process.md (hunk 3 above)
Comment (kevjumba) [reply docs/project/release-process.md]: fixed

File: docs/project/release-process.md (hunk 2 above)
Comment (kevjumba) [reply docs/project/release-process.md]: fixed

File: docs/project/release-process.md [hunk 5]
```diff
@@ -4,60 +4,27 @@
...
+5. All of the jobs should succeed besides the UI job which needs to be released separately. Ping a maintainer on Slack to run the UI release manually.
+6. Try to install the feast release in your local environment and test out the `feast init` -> `feast apply` workflow to verify as a sanity check that the release worked correctly.
 
 ### Flag Breaking Changes & Deprecations
 
 It's important to flag breaking changes and deprecation to the API for each release so that we can maintain API compatibility.
 
-Developers should have flagged PRs with breaking changes with the `compat/breaking` label. However, it's important to double check each PR's release notes and contents for changes that will break API compatibility and manually label `compat/breaking` to PRs with undeclared breaking changes. The change log will have to be regenerated if any new labels have to be added.
+Developers should have flagged PRs with breaking changes with the `compat/breaking` label. However, it's important to double check each PR's release notes and contents for changes that will break API compatibility and manually label `compat/breaking` to PRs with undeclared breaking changes.
```
Comment (felixwang9817) [inline docs/project/release-process.md:30]: is this true? I thought we didn't use `compat/breaking`, instead we used the `!` notation as in this PR: https://github.com/feast-dev/feast/pull/2418

File: docs/project/release-process.md (hunk 4 above)
Comment (felixwang9817) [reply docs/project/release-process.md:15]: ah okay I misunderstood

File: docs/project/release-process.md [hunk 6]
```diff
@@ -4,60 +4,27 @@
...
-   * Make sure that release notes and breaking changes are present.
+### Pre-release Verification (Verification that wheels are built correctly)
+
+1. Merge upstream master changes into your fork.
+2. Create a tag manually for the release on your fork. For example, if your release doing a release for version 0.22.0, create a tag by doing the following.
+   - Checkout master branch and run `git tag v0.22.0`.
+   - Run `git push --tags` to push the tag to remote.
+3. Access the `Actions` tab on your github UI on your fork and click the `build_wheels` action.
+4. Look for the header `This workflow has a workflow_dispatch event trigger` and click `Run Workflow` on the right.
+5. Run the branch off of the tag you just created(`v0.22.0` in this case) and verify that the workflow worked.
```
Comment (felixwang9817) [inline docs/project/release-process.md:15]: also do you mean `Run the workflow`?

File: docs/project/release-process.md (hunk 5 above)
Comment (kevjumba) [reply docs/project/release-process.md:30]: im not sure but ill just delete.

File: docs/project/release-process.md (hunk 6 above)
Comment (kevjumba) [reply docs/project/release-process.md:15]: fixed

File: docs/project/release-process.md [hunk 7]
```diff
@@ -4,60 +4,21 @@
...
-
-### Flag Breaking Changes & Deprecations
-
-It's important to flag breaking changes and deprecation to the API for each release so that we can maintain API compatibility.
-
-Developers should have flagged PRs with breaking changes with the `compat/breaking` label. However, it's important to double check each PR's release notes and contents for changes that will break API compatibility and manually label `compat/breaking` to PRs with undeclared breaking changes. The change log will have to be regenerated if any new labels have to be added.
+### Pre-release Verification (Verification that wheels are built correctly)
+
+1. Merge upstream master changes into your fork.
+2. Create a tag manually for the release on your fork. For example, if your release doing a release for version 0.22.0, create a tag by doing the following.
```
Comment (felixwang9817) [inline docs/project/release-process.md:10]: "if you are doing"

Comment (kevjumba) [reply docs/project/release-process.md:10]: Wait i think doing the following is what I meant;.

Comment (felixwang9817) [reply docs/project/release-process.md:10]: oh I meant `if your release doing a release` -> `if you are doing a release`

Comment (kevjumba) [reply docs/project/release-process.md:10]: fixed

**Review by felixwang9817 — APPROVED** [review]
/lgtm

File: docs/project/release-process.md [hunk 8]
```diff
@@ -4,60 +4,27 @@
...
-### Flag Breaking Changes & Deprecations
-
-It's important to flag breaking changes and deprecation to the API for each release so that we can maintain API compatibility.
-
-Developers should have flagged PRs with breaking changes with the `compat/breaking` label. However, it's important to double check each PR's release notes and contents for changes that will break API compatibility and manually label `compat/breaking` to PRs with undeclared breaking changes. The change log will have to be regenerated if any new labels have to be added.
+### Pre-release Verification (Verification that wheels are built correctly) for minor release.
+1. Merge upstream master changes into your **fork**. Make sure you are running the workflow off of your fork!
+2. Create a tag manually for the release on your fork. For example, if you are doing a release for version 0.22.0, create a tag by doing the following.
+   - Checkout master branch and run `git tag v0.22.0`.
+   - Run `git push --tags` to push the tag to remote.
```
Comment (achals) [inline docs/project/release-process.md:11]: To push the tag to your forks master branch.

File: docs/project/release-process.md [hunk 9]
```diff
@@ -4,60 +4,27 @@
...
+3. Access the `Actions` tab on your github UI on your fork and click the `build_wheels` action.
+4. Look for the header `This workflow has a workflow_dispatch event trigger` and click `Run Workflow` on the right.
+5. Run the workflow off of the tag you just created(`v0.22.0` in this case) and verify that the workflow worked.
```
Comment (achals) [inline docs/project/release-process.md:14]: What kind of verification?

File: docs/project/release-process.md [hunk 10]
```diff
@@ -4,60 +4,27 @@
...
+
+### Pre-release Verification (Verification that wheels are built correctly) for patch release.
+1. Check out the branch of your release (e.g `v0.22-branch` on your local **fork**) and push this to your fork (`git push -u origin <branch>`).
+2. Cherry pick commits that are relevant to the patch release onto your forked branch.
+3. Checkout the release branch and add a patch release tag (e.g `v0.22.1`) by running `git tag <tag>`.
+4. Push tags to your origin branch with `git push origin <tag>`.
+5. Kick off `build_wheels` workflow in the same way as is detailed in the last section on of the patch release tag.
+
+### Release
```
Comment (achals) [inline docs/project/release-process.md:23]: What stuff does this workflow do? Python release? Java release? UI release? Helm charts? Docker images?

File: docs/project/release-process.md [hunk 11]
```diff
@@ -4,60 +4,27 @@
...
-
-It's important to flag breaking changes and deprecation to the API for each release so that we can maintain API compatibility.
-
-Developers should have flagged PRs with breaking changes with the `compat/breaking` label. However, it's important to double check each PR's release notes and contents for changes that will break API compatibility and manually label `compat/breaking` to PRs with undeclared breaking changes. The change log will have to be regenerated if any new labels have to be added.
+### Pre-release Verification (Verification that wheels are built correctly) for minor release.
+1. Merge upstream master changes into your **fork**. Make sure you are running the workflow off of your fork!
+2. Create a tag manually for the release on your fork. For example, if you are doing a release for version 0.22.0, create a tag by doing the following.
+   - Checkout master branch and run `git tag v0.22.0`.
+   - Run `git push --tags` to push the tag to remote.
+3. Access the `Actions` tab on your github UI on your fork and click the `build_wheels` action.
```
Comment (achals) [inline docs/project/release-process.md:12]: What does this workflow do? What does it build? What combinations?

File: docs/project/release-process.md (hunk 8 above)
Comment (kevjumba) [reply docs/project/release-process.md:11]: Fixed

File: docs/project/release-process.md (hunk 9 above)
Comment (kevjumba) [reply docs/project/release-process.md:14]: clarified.

File: docs/project/release-process.md (hunk 10 above)
Comment (kevjumba) [reply docs/project/release-process.md:23]: Clarified

File: docs/project/release-process.md (hunk 11 above)
Comment (kevjumba) [reply docs/project/release-process.md:12]: Fixed.

File: setup.py [hunk 12]
```diff
@@ -430,7 +430,7 @@ def build_extension(self, ext: Extension):
         destination = os.path.dirname(os.path.abspath(self.get_ext_fullpath(ext.name)))
         subprocess.check_call(["go", "install", "golang.org/x/tools/cmd/goimports"],
                               env={"PATH": bin_path, **go_env})
-        subprocess.check_call(["go", "get", "github.com/go-python/gopy@v0.4.0"],
+        subprocess.check_call(["go", "get", "github.com/go-python/gopy@v0.4.4"],
```
Comment (achals) [inline setup.py:433]: Have we updated the build process to pass in the new flags? If not, this change is incomplete (and should be in a different PR anyway).

**Review by achals — CHANGES_REQUESTED** [review]

File: setup.py (hunk 12 above)
Comment (kevjumba) [reply setup.py:433]: yup

**Review by felixwang9817 — APPROVED** [review]
/lgtm
